File: CoreLib/src/main/java/org/hexnibble/corelib/commands/rc/InstantRC.java
package org.hexnibble.corelib.commands.rc;

public class InstantRC extends RC {
   Runnable function;

   public InstantRC(Runnable function) {
      this.function = function;
   }

   @Override
   protected void processCommand() {
      function.run();
      setCommandStatus(COMMAND_STATUS.COMMAND_SUCCESSFULLY_COMPLETED);
   }
}

File: CoreLib/src/main/java/org/hexnibble/corelib/commands/rc/InstantRC.kt
package org.hexnibble.corelib.commands.rc

/**
 * Executes a Runnable and immediately return.
 * @param function The Runnable to run.
 * @param commandID The ID of the command.
 */
class InstantRC @JvmOverloads constructor(
  private val function: Runnable,
  commandID: String = "InstantRC"
) : RC(commandID) {
  override fun processCommand() {
    function.run()
    commandStatus = COMMAND_STATUSES.COMMAND_SUCCESSFULLY_COMPLETED
  }
}
